web-console/src/views/segments-view/segments-view.tsx

```diff
@@ -114,7 +116,7 @@ interface SegmentQueryResultRow {
 export class SegmentsView extends React.PureComponent<SegmentsViewProps, SegmentsViewState> {
   private segmentsSqlQueryManager: QueryManager<QueryAndSkip, SegmentQueryResultRow[]>;
   private segmentsJsonQueryManager: QueryManager<any, SegmentQueryResultRow[]>;
-
+  private previousTableState: any;
```

Comment:
you should not need this

Reply:
Why I am using that is because when you click on the interval button you have to call fetch data() because there is no onPivotBy for the react table or anything like that. You need the state of the table though to get the number of segments, and I'm not sure how to get that from outside the table. I looked at the documentation and there are like accessors for you to set state but not to return state, I also though about using a ref but typescript didn't like that. How would you solve that problem ?